repos/fdbt-site/src/data/auroradb.ts

```diff
@@ -705,6 +730,82 @@ export const getPassengerTypeByNameAndNocCode = async (
     }
 };
 
+export const getSinglePassengerTypeByNameAndNationalOperatorCode = async (
+    nationalOperatorCode: string,
+    name: string,
+    isGroup: boolean,
+): Promise<SinglePassengerType | undefined> => {
+    logger.info('', {
+        context: 'data.auroradb',
+        message: 'retrieving passenger type for a given national operator code and name',
+        nationalOperatorCode,
+        name,
+        isGroup,
+    });
+
+    try {
+        const queryInput = `
+            SELECT id, name, contents
+            FROM passengerType
+            WHERE nocCode = ? AND name = ? AND isGroup = ?`;
+
+        const queryResults = await executeQuery<{ id: number; name: string; contents: string }[]>(queryInput, [
+            nationalOperatorCode,
+            name,
+            isGroup,
+        ]);
+
+        if (queryResults.length > 1) {
+            throw new Error("Didn't expect more than one passenger type with the same national operator code and name");
+        }
+
+        const data = queryResults[0];
+
+        return data
+            ? ({
+                  id: data.id,
+                  name: data.name,
+                  passengerType: JSON.parse(data.contents),
+              } as SinglePassengerType)
+            : undefined;
+    } catch (error) {
+        throw new Error(`Could not retrieve passenger type by national operator code and name from AuroraDB: ${error}`);
+    }
+};
+
+export const getPassengerTypeById = async (id: number): Promise<SinglePassengerType | undefined> => {
+    logger.info('', {
+        context: 'data.auroradb',
+        message: 'retrieving passenger type for a given id',
+        id: id,
+    });
+
+    try {
+        const queryInput = `
+            SELECT id, name, contents
+            FROM passengerType
+            WHERE id = ?`;
+
+        const queryResults = await executeQuery<{ id: number; name: string; contents: string }[]>(queryInput, [id]);
+
+        if (queryResults.length > 1) {
+            throw new Error("Didn't expect more than one passenger type with the same id");
+        }
+
+        const data = queryResults[0];
+
+        return data
+            ? ({
+                  id: data.id,
+                  name: data.name,
+                  passengerType: JSON.parse(data.contents),
+              } as SinglePassengerType)
```

Comment:
If possible it would be better to cast the passengerType parsed string only, as casting the whole thing would hide where there were type inconsistencies in the larger structure